### ISSUE TITLE
Embedding impress into application

### ISSUE THREAD

**Issue by cimpok** [impress/impress.js#751]
I am struggling trying to import impress.js into a Vue application, it seems to me that I am just missing  a single export function for the plugin feature. My goal would be a setup to use impress.js like reveal.js is used here [https://codepen.io/lucafaggianelli/pen/KQKrVr]()
Could you please give me some clue where to put the missing exports in the code.

My other question is: what is the right approach to use impress not full-screen, place it into an iframe or modify the main div's size in the code?
Sorry for asking instructions here, please consider this as a feature request.

**Comment by henrikingo** [comment]
Impress fills and scales+transforms the entire window and also adds a lot of classes to body. I always use an iframe to embed it, I don't see how anything else makes sense.

Otoh the opposite is possible: If you have another javascript framework that plays nicely inside a div, you can embed such all you want into individual slides.

**Comment by cimpok** [comment]
Thank you for your kind comment. And also thank you Henrik for developing and taking care of impress, which I found when I was looking for a possible add-on to my kiosk/signage/video solution running ads/company info/etc on remotely managed screens. I was looking for a framework to be able to overlay some user-editable text on a transparent-background layer over my content. I was only hoping for a simple fade-in/out effect, but I got impressed* No rhyme here, just a pun (intended :-). I knew about prezi, and was not hoping for an open-source solution matching and even beating it.

By embedding I mean two scenarios:
1. a full screen layer running on top of HTML5 videos or still images - my kiosk players are low-end machines, that's why I am looking for impress to do the texts only (I may still have to resist anything fancier that fading and sliding. As I observed zooming and especially the rotation needs too much computing power not available in these machines. I could however rotate and zoom a running video in a slide on my develepoment machine wonderfully)  Btw. did you came across a html5 renderer that would output the whole animation (ie. contents of an iframe) in a ready-to-play-anywhere video format?
2. a preview on the admin screen running on part-screen, iframe (or div* - was my question, thank you for clarifying it)

* stripping off all user-interaction related addEventListeners attached to body in your code (I don't need user input, there isn't any on these players, the slides sequence by their timers only) and adding some functions to the API I managed to run impress in a div. It is just a proof of concept now, at the first tries I had some issues with the sizing, but it is doable.
I would not call this (mostly removing functionality) for a 'development' at this phase, but with a more mature solution, after adding the missing features I would be honored to be able to post here a fork of an embeddable impress player (maybe with a video file/stream output?) some time, if you don't object.
